app/src/main/java/com/example/printer/printer/PrinterService.kt

```diff
@@ -423,10 +418,12 @@ class PrinterService(private val context: Context) {
                             submissionTime = System.currentTimeMillis(),
                             state = com.example.printer.queue.PrintJobState.PENDING
                         )
-                        // Use a new coroutine scope to avoid Compose conflicts
-                        kotlinx.coroutines.runBlocking(kotlinx.coroutines.Dispatchers.IO) { 
+                        // Direct suspend call with timeout (already in Ktor coroutine context)
+                        kotlinx.coroutines.withTimeout(30000) { // 30 second timeout
```

Comment:
If you use `30.seconds.inWholeMilliseconds` instead of `30000`, you'll get the same timeout without the need for a comment.

Reply:
Same for the other hardcoded timeouts throughout this file.